packages/bot-skeleton/src/services/api/api-base.js

```diff
@@ -32,10 +34,20 @@ class APIBase {
         if (this.api) this.api.disconnect();
     }
 
+    stopBotIfSocketDisconnected() {
+        if (this.api.connection.readyState === 3) {
+            const { stopBot, clearStat } = DBotStore.instance.run_panel;
+            stopBot();
+            clearStat(true);
+        }
+    }
+
     initEventListeners() {
         if (window) {
             window.addEventListener('online', this.reconnectIfNotConnected);
             window.addEventListener('focus', this.reconnectIfNotConnected);
+            //we can even run the bot by opening a new websocket connection here!
+            document.addEventListener('visibilitychange', () => this.stopBotIfSocketDisconnected());
```

Comment:
Can we listen to WS status change instead of listening to visibility of the tab?

Reply:
@balakrishna-deriv yes will try using this `this.api.connection.addEventListener('close', () => this.stopBotIfSocketDisconnected());`